### PR TITLE
Add ssec_key_hash and ssec_iv into the custom-metadata section of the needle

### PR DESCRIPTION
**Pull request by yosukehara**
* Add `ssec_key_hash` and `ssec_iv` into the custom-metadata section of the needle
* Format code to be more readable
* Add AUTHORS file instead of `@author`

### CODE REVIEW

File: include/leo_object_storage.hrl [hunk 1]
```diff
@@ -202,9 +202,10 @@
 %% AVS-Related
 %%--------------------------------------------------------------------
 %% AVS version strings
--define(AVS_HEADER_VSN_2_2,  <<"LeoFS AVS-2.2">>). %% leofs v0.14 - v1.0.0-pre1
--define(AVS_HEADER_VSN_2_4,  <<"LeoFS AVS-2.4">>). %% leofs v1.0.0-pre1 - current ver
--define(AVS_HEADER_VSN_TOBE, ?AVS_HEADER_VSN_2_4).
+-define(AVS_HEADER_VSN_2_2,  <<"LeoFS AVS-2.2">>). %% leofs v0.14 - v0.16
+-define(AVS_HEADER_VSN_2_4,  <<"LeoFS AVS-2.4">>). %% leofs v1.0 - v1.4
+-define(AVS_HEADER_VSN_3_0,  <<"LeoFS AVS-3.0">>). %% leofs v2.0
```
Comment (mocchira) [inline include/leo_object_storage.hrl]: No longer needed as sse-c related field will be included in the custom metadata field?

File: include/leo_object_storage.hrl [hunk 2]
```diff
@@ -228,12 +229,14 @@
                 ?AVS_HEADER_VSN_2_2 ->
                     term_to_binary({_AddrId, _Key});
                 ?AVS_HEADER_VSN_2_4 ->
+                    _Key;
+                ?AVS_HEADER_VSN_3_0 ->
                     _Key
```
Comment (mocchira) [inline include/leo_object_storage.hrl]: ditto.

File: include/leo_object_storage.hrl [hunk 3]
```diff
@@ -228,12 +229,14 @@
...
             end
         end).
 
 -define(AVS_HEADER_VSN,     <<?AVS_HEADER_VSN_TOBE/binary,13,10>>).
--define(AVS_PART_OF_HEADER, <<"CHKSUM:128,KSIZE:16,BLEN_MSIZE:32,DSIZE:32,OFFSET:64,ADDRID:128,CLOCK:64,TIMESTAMP:42,DEL:1,BUF:437,CHUNK_SIZE:32,CHUNK_NUM:24,CHUNK_INDEX:24",13,10>>).
+-define(AVS_PART_OF_HEADER, <<"CHKSUM:128,KSIZE:16,BLEN_MSIZE:32,DSIZE:32,OFFSET:64,ADDRID:128,CLOCK:64,TIMESTAMP:42,DEL:1,BUF:437,CHUNK_SIZE:32,CHUNK_NUM:24,CHUNK_INDEX:24,SSEC_KEY_HASH:32,SSEC_IV",13,10>>).
```
Comment (mocchira) [inline include/leo_object_storage.hrl]: ditto. SSEC_KEY_HASH:32,SSeC_IV are no longer needed here?

File: src/leo_object_storage_haystack.erl [hunk 4]
```diff
@@ -694,13 +691,18 @@ create_needle(#?OBJECT{addr_id = AddrId,
                        offset = Offset,
                        timestamp = Timestamp,
                        checksum = Checksum,
-                       del = Del}) ->
+                       del = Del
+                       %% ssec_key_hash = SSEC_KeyHash,
+                       %% ssec_iv = SSEC_IV
```
Comment (mocchira) [inline src/leo_object_storage_haystack.erl]: Remove unnecessary comments.

File: src/leo_object_storage_haystack.erl [hunk 5]
```diff
@@ -719,6 +721,9 @@ create_needle(#?OBJECT{addr_id = AddrId,
                  CSize:?BLEN_CHUNK_SIZE,
                  CNum:?BLEN_CHUNK_NUM,
                  CIndex:?BLEN_CHUNK_INDEX,
+                 %% @TODO
+                 %% SSEC_KeyHash:?BLEN_SSEC_KEY_HASH,
+                 %% SSEC_IV:?BLEN_SSEC_IV,
```
Comment (mocchira) [inline src/leo_object_storage_haystack.erl]: Ditto.

File: include/leo_object_storage.hrl (hunk 3 above)
Comment (yosukehara) [reply include/leo_object_storage.hrl]: @mocchira Right, `SSEC_KEY_HASH:32,SSEC_IV` are not necessary. I'll rollback this line.

**Comment by yosukehara** [comment]
@mocchira PTAL

**Pull request merged**